app/views/unattended/provisioning_templates/cloud_init/preseed_cloud-init-autoinstall-user-data.erb

```diff
@@ -0,0 +1,46 @@
+<%#
+kind: provision
+name: preseed_cloud-init-autoinstall-user-data
+model: ProvisioningTemplate
+oses:
+- Ubuntu 20.04 live-server and higher
+test_on:
+- ubuntu 20.04.3 live-server
+-%>
+<%-
+realname_to_create = host_param('realname_to_create')
+username_to_create = host_param('username_to_create')
+userpassword_to_create = host_param('userpassword_to_create')
+custom_sources_list = host_param('custom_sources_list')
+-%>
+#cloud-config
+autoinstall:
+<%= indent 2 do snippet_if_exists(template_name + " custom bootcmd") end -%>
+  apt:
+    geoip: false
+    preserve_sources_list: false
+<%- if custom_sources_list -%>
+<%= indent 4 do snippet 'preseed_cloud-init-autoinstall-user-data custom sources_list' end -%>
+<%- else -%>
```

Comment:
Do you really need this? I am not pulling this into my PR because it's clunky, I would suggest that if you need that you implement this via a host parameter that could be in the form of like: `amd64,i386: http://xyz, default: http://xyz` and it would have a default value and then here instead of if-else you would simple break the string into elements and iterate over it (split by , and split by colon).